apps/language_server/lib/language_server/providers/completion.ex

```diff
@@ -294,19 +311,26 @@ defmodule ElixirLS.LanguageServer.Providers.Completion do
            origin: origin,
            metadata: metadata
          },
-         context
+         context,
+         options
        ) do
     def_str = if(context[:def_before] == nil, do: "def ")
 
-    full_snippet = "#{def_str}#{snippet(name, args, arity)} do\n\t$0\nend"
+    insert_text =
+      if Keyword.get(options, :snippets_supported, false) do
+        "#{def_str}#{snippet(name, args, arity)} do\n\t$0\nend"
+      else
+        "#{def_str}#{name}"
+      end
```

Comment:
Since this exact snippet is used twice it would be good to extract it into a function